File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/DeviceRepository.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.Device;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;
import java.util.Optional;

public interface DeviceRepository extends MongoRepository<Device, String> {

    Optional<Device> findOneByToken(String token);

    List<Device> findByUserId(String userId);

    Optional<Device> findOneByIdOrToken(String id, String token);
}

File: izinga-commons/src/main/kotlin/commons/repo/DeviceRepository.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.Device
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.*

interface DeviceRepository : MongoRepository<Device?, String?> {
    fun findOneByToken(token: String?): Optional<Device?>?
    fun findByUserId(userId: String?): List<Device?>?
    fun findOneByIdOrToken(id: String?, token: String?): Optional<Device?>?
}